### DISCUSSION TITLE
JWTExpired: "exp" claim timestamp check failed (with a fresh JWT token)

### DISCUSSION THREAD

**Discussion by calboru** [Web3Auth/Web3Auth#585] [Help]
Hi, 
I get "JWTExpired: "exp" claim timestamp check failed" error when validating JWT token on server side.
CODE
------
      const jwks = jose.createRemoteJWKSet(
        new URL("https://api.openlogin.com/jwks")
      );

      // Verify the JWT using Web3Auth's JWKS
      const jwtDecoded = await jose.jwtVerify(idToken, jwks, {
        algorithms: ["ES256"],
      });
------
JWT Token:
=========
eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6IlRZT2dnXy01RU9FYmxhWS1WVlJZcVZhREFncHRuZktWNDUzNU1aUEMwdzAifQ.eyJpYXQiOjE2NjE5NzkyMDcsImF1ZCI6IkJQdVNWOFFhLU5XTWxJa3F2WXJvMkJEeUtNaG1vUkJuTjlpT205WmMxalhfekFYVk8wcmhzSmxDcmQ4YmJkbWpoOUk3WHBKXzZLX1p3QTVIaTc5UlNIVSIsIm5vbmNlIjoiMDM1ZDczNTc1ZmQ1OGUyMjFkM2MzOTUzZTg5YTJjOWQzNjc1OTkzYjYyZTNkNDRhMDA0YjAwMDI0MDExOWM1NGY3IiwiaXNzIjoiaHR0cHM6Ly9hcGkub3BlbmxvZ2luLmNvbSIsIndhbGxldHMiOlt7InB1YmxpY19rZXkiOiIwMjI1MGQ0MzliMjdkN2ZmNjY1Y2ZhMzcyMWU0MzNjNzcyN2Y2YzcxNTBkOTAxZjYyODc3MjFmZmI2ZWNhMGQ2YmEiLCJ0eXBlIjoid2ViM2F1dGhfYXBwX2tleSIsImN1cnZlIjoic2VjcDI1NmsxIn1dLCJlbWFpbCI6Im9rYW5ncmF5d29sZkBnbWFpbC5jb20iLCJuYW1lIjoiT2thbiBHcmF5d29sZiIsInByb2ZpbGVJbWFnZSI6Imh0dHBzOi8vcGJzLnR3aW1nLmNvbS9wcm9maWxlX2ltYWdlcy8xNTE5NjcwOTI3ODAxNDM0MTE0L0NUZk1YcUh1X25vcm1hbC5qcGciLCJ2ZXJpZmllciI6InRvcnVzIiwidmVyaWZpZXJJZCI6InR3aXR0ZXJ8MTUxNTQxMjYwMTgyNzcwMDc0MSIsImFnZ3JlZ2F0ZVZlcmlmaWVyIjoidGtleS1hdXRoMC10d2l0dGVyLWxyYyIsImV4cCI6MTY2MjA2NTYwN30.uVoYs8mo3e51mPEHV2UlzYgvQY7zlDvDo-NI0bVcRC8pSp8Z2SkfYvpSov9ITqJL5nBFR6Gp93B-LVyJ0G4m-A

**Comment by shahbaz17** [comment]
parsedToken.idToken is the idToken you receive in getUserInfo? Or is it parsed?

**Reply by calboru** [reply]
Yes the idToken from getUserInfo

**Reply by calboru** [reply]
By the way this is social login.

**Reply by calboru** [reply]
Hi Shahbaz, as of today it is still not working. Now I am being forwarded to the support chat screen.

**Reply by calboru** [reply]
I had to clear Safari's (Iphone) cache. Now it is working properly. Why?

**Reply by shahbaz17** [reply]
Maybe it was using the old `idToken` that has already expired.

**Comment by shahbaz17** [comment]
I am marking this resolved, as the code you shared was accurate and can be used to verify the `id_token`.

**Comment by grabbou** [comment]
Unfortunately, this started happening today for me too. The login was successful, but the expired token was returned. Everything worked till yesterday, together with new token being issued on subsequent logins. No code changed since then. Anyway to debug this?
